utils/db_utils.py

```diff
@@ -437,58 +437,51 @@ def add_user_stats(user_data, batch_size=5):
                     user['manufacturer'] = profile_data.get('manufacturer')
                     user['app_version'] = profile_data.get('client_app_version')
                     user['os_version'] = profile_data.get('client_os_version')
                     user['phone_lang'] = profile_data.get('phone_lang')
-    
-                if total_trips > 0:
-                    ts = esta.TimeSeries.get_time_series(user_uuid)
-                    first_trip_ts = ts.get_first_value_for_field(
-                        key='analysis/confirmed_trip',
-                        field='data.end_ts',
-                        sort_order=pymongo.ASCENDING
-                    )
-                    if first_trip_ts != -1:
-                        user['first_trip'] = arrow.get(first_trip_ts).format(time_format)
-    
-                    last_trip_ts = ts.get_first_value_for_field(
-                        key='analysis/confirmed_trip',
-                        field='data.end_ts',
-                        sort_order=pymongo.DESCENDING
-                    )
-                    if last_trip_ts != -1:
-                        user['last_trip'] = arrow.get(last_trip_ts).format(time_format)
-    
-                    last_call_ts = ts.get_first_value_for_field(
-                        key='stats/server_api_time',
-                        field='data.ts',
-                        sort_order=pymongo.DESCENDING
-                    )
-                    if last_call_ts != -1:
-                        user['last_call'] = arrow.get(last_call_ts).format(time_format)
-                
+
+                    
+                    # Assign newly stored statistics to the user dictionary
+                    user['total_trips'] = profile_data.get('total_trips', 0)
+                    user['labeled_trips'] = profile_data.get('labeled_trips', 0)
+                    
+                    # Retrieve and assign pipeline range
+                    pipeline_range = profile_data.get('pipeline_range', {})
+                    user['pipeline_start_ts'] = pipeline_range.get('start_ts')
+                    user['pipeline_end_ts'] = pipeline_range.get('end_ts')
+                    
+                    # Retrieve and assign last API call timestamp
+                    user['last_call'] = profile_data.get('last_call')
+                    
+                    # Optional: Format the last_call timestamp for readability
+                    if user['last_call']:
+                        user['formatted_last_call'] = arrow.get(user['last_call']).format('YYYY-MM-DD HH:mm:ss')
+                    else:
+                        user['formatted_last_call'] = None
+                else:
+                    # Handle cases where profile_data is not found
+                    logging.warning(f"No profile data found for user_uuid: {user_uuid}")
+                    
+                    # Assign default values to ensure the user dictionary remains consistent
```

Comment:
I noticed that you already provided default values in the code above using `profile_data.get()` to handle cases where fields are missing from `profile_data`.

Then, you have this else block to handle when `profile_data` is None, and you give the same default values.

I think you can simpify this by setting `profile_data` to `{}` (an empty dict) if it is `None`. then you don't need the if-else block

This is better because i) it is more concise and you do not repeat the default values; ii) the code is flatter (less things nested in an if-else statement)

Reply:
@JGreenlee Addressed